src/components/inputs/checkbox-input/checkbox-input.js

```diff
@@ -1,12 +1,80 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
+import { css } from '@emotion/core';
+import vars from '../../../../materials/custom-properties';
 import filterDataAttributes from '../../../utils/filter-data-attributes';
 import getFieldId from '../../../utils/get-field-id';
 import Text from '../../typography/text';
 import Spacings from '../../spacings';
 import Icons from './icons';
-import styles from './checkbox-input.mod.css';
+
+const getLabelStyles = (isDisabled, hasError) => {
+  if (isDisabled) {
+    return css`
+      cursor: not-allowed;
+    `;
+  }
+  if (hasError) {
+    return css``;
+  }
+  return css`
+    cursor: pointer;
+    &:hover svg [id$='borderAndContent'] > [id$='border'] {
+      stroke: ${vars.borderColorInputFocus};
+    }
+  `;
+};
+
+const getCheckboxWrapperStyles = (isDisabled, hasError, isHovered) => {
+  const baseStyles = css`
+    display: flex;
+    align-items: center;
+    svg [id$='borderAndContent'] > [id$='border'] {
+      stroke: 1px ${vars.borderColorInputPristine} solid;
+    }
+    svg [id$='borderAndContent'] > [id$='content'] {
+      fill: ${vars.borderColorInputFocus};
+    }
+  `;
+  if (isDisabled) {
+    return [
+      baseStyles,
+      css`
+        svg [id$='borderAndContent'] > [id$='content'] {
+          fill: ${vars.fontColorDisabled};
+        }
+        svg [id$='borderAndContent'] > [id$='border'] {
+          stroke: ${vars.borderColorInputDisabled};
+        }
+      `,
+    ];
+  }
+  if (hasError) {
+    return [
+      baseStyles,
+      css`
+        svg [id$='borderAndContent'] > [id$='content'] {
+          fill: ${vars.borderColorInputError};
+        }
+
+        svg [id$='borderAndContent'] > [id$='border'] {
+          stroke: ${vars.fontColorError};
+        }
+      `,
+    ];
+  }
+  if (isHovered && !isDisabled && !hasError) {
+    return [
+      baseStyles,
+      css`
+        svg [id$='borderAndContent'] > [id$='border'] {
+          stroke: ${vars.borderColorInputFocus};
+        }
+      `,
+    ];
+  }
+  return baseStyles;
+};
```

Comment:
So before we moved them into a .styles. Then we inline them. Now we have them in fns above. What do we prefer. Should we settle in a way - at least for fns in .styles or above?

Reply:
From my point of view so far:
- simple styles can be "inlined" so it's easier to read them
- if the style requires some "complex" conditionals, move the style into a function
- if you start having a lot of functions that pollutes the component, move them into a `styles` file

So for me it's more about readability. As long as the component is readable, it's fine to have the styles within the same file, otherwise move it to a separate file.

In this particular case I guess we can move the functions into a separate file

Reply:
Yes, I agree with the loose rule of thumb and also that in this case they should be moved.